edx_sailthru/import_users_activation_codes/import_activation_codes.py

```diff
@@ -0,0 +1,136 @@
+import argparse
+import logging
+
+import MySQLdb
+import os
+import pandas
+import sys
+from sailthru.sailthru_client import SailthruClient
+
+logger = logging.getLogger(__name__)
+
+
+def setup_logging():
+    logger.setLevel(logging.INFO)
+    logging.basicConfig(format='%(asctime)s - %(levelname)s - %(message)s')
+
+
+def get_list_of_inactive_users(args):
+    edxapp_db = MySQLdb.connect(host=args.mysql_host,
+                                user=args.mysql_user,
+                                passwd=args.mysql_pass,
+                                db=args.db)
+
+    if args.is_active is None:
+        active_query = ""
+    elif args.is_active:
+        active_query = "AND au.is_active=1"
+    else:
+        active_query = "AND au.is_active=0"
+
+    sql = """SELECT au.email, au.is_active, ar.activation_key
+    FROM auth_user au
+    INNER JOIN auth_registration ar
+    ON au.id=ar.user_id
+    """ + active_query
+
+    users_data = pandas.read_sql_query(sql, edxapp_db)
+
+    return users_data
+
+
+def update_user_on_sailthru(users_data, client):
+    logger.info("Starting updating users on sailthru")
+
+    for index, user in users_data.iterrows():
```

Comment:
Have you tried to use their batch/job interface recommended in the ticket?
you will be making millions of API calls this way. That's quite inefficient.

Reply:
I read the docs related to it and it is stated that `If you want to apply the same update to all of the users specified, you should pass the update parameter.` So I believe we are updating every user with his/her own `activation_code` so we can do a bulk update.